eduNEXT/eox-theming
# perf: add redwood compatibility

## Description

This PR aims to upgrade the plugin and workflow requirements, update the README and make it clearer, and fix a bug with the `footer_override` rendering [Issue 9](https://github.com/eduNEXT/eox-release/issues/9). 

## Testing instructions

1. Create a redwood environment and install distro `pip install git+https://github.com/eduNEXT/tutor-contrib-edunext-distro`
2. Install the MFE tutor plugin and enable it
3. Add the following settings in the `config.yml`
``` yml
DISTRO_EOX_THEMING_DPKG:
  domain: github.com
  index: git
  name: eox-theming
  path: eduNEXT
  private: false
  protocol: https
  repo: eox-theming
  variables:
    development:
      EOX_THEMING_CONFIG_SOURCES:
      - from_eox_tenant_microsite_v2
      - from_django_settings
    production:
      EOX_THEMING_CONFIG_SOURCES:
      - from_eox_tenant_microsite_v2
      - from_django_settings
  version: bc/redwood-compatibility
OPENEDX_EXTRA_PIP_REQUIREMENTS:
- eox-tenant
DISTRO_THEMES:
- domain: github.com
  name: ednx-test-themes
  path: eduNEXT
  protocol: ssh
  repo: ednx-test-themes
  version: bc/test-footer
DISTRO_THEMES_NAME:
- test-1
- test-2
- test-3
DISTRO_THEMES_ROOT: /openedx/themes
DISTRO_THEME_DIRS:
- /openedx/themes/ednx-test-themes/edx-platform
```
4. Clone the test themes folder with `tutor distro enable-themes`
5. Add the testing themes running and init the environment running `tutor dev launch`
6. Create a new tenant and add this info to the tenant configs:
``` json
{
    "EDNX_USE_SIGNAL": true,
    "ENABLE_AUTHN_MICROFRONTEND": false,
    "ENABLE_ACCOUNT_MICROFRONTEND": false,
    "THEME_OPTIONS": {
        "scripts": {
            "/login": [
                {
                    "content": "alert('This is a test for the inline script');",
                    "type": "inline"
                }
            ]
        },
        "theme": {
            "grandparent": "test-3",
            "name": "test-1",
            "parent": "test-2"
        }
    },
    "course_org_filter": [
        "OpenedX"
    ],
    "footer_override": "<div style=\"width: 100%;\"><div style=\"position: relative; padding-bottom: 41.67%; padding-top: 0; height: 0;\"><iframe title=\"Footer X\" frameborder=\"0\" width=\"1200\" height=\"500\" style=\"position: absolute; top: 0; left: 0; width: 100%; height: 100%;\" src=\"https://view.genially.com/66464d3334225f0015c0dc0e\" type=\"text/html\" allowscriptaccess=\"always\" allowfullscreen=\"true\" scrolling=\"yes\" allownetworking=\"all\"></iframe> </div> </div>",
    "header_langselector": true,
    "header_links": [
        {
            "target": "_self",
            "txt": "Courses",
            "url": "/courses"
        },
        {
            "class": "mx-2",
            "target": "_self"
        }
    ],
    "released_languages": "en,vi,es-419"
}
```
7. Go to your admin panel and create a waffle flag `learner_home_mfe.enabled` with the property `Everyone` in false to disable the Authn MFE
8. The test cases from [this file](https://docs.google.com/document/d/18urdJPitAu1lUBvYdQcGXm4WRDbKD2eDf4TTGgrlzZU/edit)

To test the solved bug:
1. Add the demo course `tutor dev do importdemocourse`
2. Enable the Wiki going to http://apps.local.edly.io:2001/course-authoring/course/course-v1:OpenedX+DemoX+DemoCourse/pages-and-resources/wiki/settings
3. Then go to the course Wiki http://local.edly.io:8000/wiki/OpenedX.DemoX.DemoCourse/
4. You should see the footer overridden with the code of 'footer_override' from the tenant settings

## JIRA ISSUES
- [DS-1067](https://edunext.atlassian.net/browse/DS-1067)
- [DS-1098](https://edunext.atlassian.net/browse/DS-1098)
- [DS-1012](https://edunext.atlassian.net/browse/DS-1012)

[DS-1067]: https://edunext.atlassian.net/browse/DS-1067?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
